core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -1126,8 +1132,12 @@ class HttpGoogleServicesDAO(
     cloudResourceManager
   }
 
-  private def getUserCredential(userInfo: UserInfo): Credential = {
-    new GoogleCredential().setAccessToken(userInfo.accessToken.token).setExpiresInSeconds(userInfo.accessTokenExpiresIn)
+  private def getUserCredential(userInfo: UserInfo): Option[Credential] = {
+    // Use the Google token if present to build the credential
+    val tokenOpt = if (userInfo.isB2C) userInfo.googleAccessTokenThroughB2C else Some(userInfo.accessToken)
```

Comment:
❓ Any way to get test coverage on this fallback logic?

Reply:
Good call - this is probably unit-testable. Will add.